core/tests/behavior/list.rs

```diff
@@ -84,6 +86,95 @@ pub async fn test_list_dir(op: Operator) -> Result<()> {
     Ok(())
 }
 
+/// List dir with metakey
+pub async fn test_list_dir_with_metakey(op: Operator) -> Result<()> {
+    let parent = uuid::Uuid::new_v4().to_string();
+    let path = format!("{parent}/{}", uuid::Uuid::new_v4());
+    debug!("Generate a random file: {}", &path);
+    let (content, size) = gen_bytes();
+
+    op.write(&path, content).await.expect("write must succeed");
+
+    let mut obs = op
+        .lister_with(&format!("{parent}/"))
+        .metakey(
+            Metakey::Mode
+                | Metakey::CacheControl
+                | Metakey::ContentDisposition
+                | Metakey::ContentLength
+                | Metakey::ContentMd5
+                | Metakey::ContentRange
+                | Metakey::ContentType
+                | Metakey::Etag
+                | Metakey::LastModified
+                | Metakey::Version,
```

Comment:
We could use like:
```rust
use flagset::{FlagSet, flags};

flags! {
    pub enum MetaKey: u64 {
        Mode,

        Foo,
        Bar
    }
}

fn main() {
    let set: FlagSet<MetaKey> = FlagSet::full();

    assert_eq!(MetaKey::Bar | MetaKey::Foo, set & !MetaKey::Mode);
}
```

So,

```rust
let full_set: FlagSet<MetaKey> = FlagSet::full();
```
Then this part could be replaced by `full_set & !MetaKey::Complete`. We wouldn't have to maintain this place so often.

Reply:
I'm guessing it's ok in test code. I prefer to keep current design of `Metakey` layout.

Reply:
Ok. No problem.